lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -272,13 +294,68 @@ static LogicalResult verifyCellOp(CellOp cell) {
 //===----------------------------------------------------------------------===//
 static LogicalResult verifyAssignOp(AssignOp assign) {
   auto parent = assign->getParentOp();
-  if (!(isa<GroupOp>(parent) || isa<WiresOp>(parent)))
+  if (!isa<GroupOp, WiresOp>(parent))
     return assign.emitOpError(
         "should only be contained in 'calyx.wires' or 'calyx.group'");
 
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ControlOp
+//===----------------------------------------------------------------------===//
+
+/// A helper function to verify that each operation in
+/// the body of a control-like operation is valid.
+static LogicalResult verifyControlLikeOpBody(Operation *op) {
+  assert(op->getNumRegions() != 0 && "The operation should have a region.");
+  auto &region = op->getRegion(0);
+  assert(region.hasOneBlock() && "The region should have one block.");
```

Comment:
I think these assertions will be guaranteed by ODS if you add the above traits traits to the ops.